src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -393,24 +399,46 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
     reset_io();
 }
 
-void ov::npuw::CompiledModel::fill_weights_bank(const std::size_t idx) {
-    LOG_VERB("Filling weights bank for Subgraph[" << idx << "]...");
-    LOG_BLOCK();
+void ov::npuw::CompiledModel::finalize_weights_bank() {
+    ov::parallel_for(m_compiled_submodels.size(), [&](std::size_t idx) {
+        auto& comp_model_desc = m_compiled_submodels[idx];
+
+        if (!comp_model_desc.replaced_by) {
+            return;
+        }
+
+        const auto real_idx = comp_model_desc.replaced_by.value();
+        auto& func_desc = m_compiled_submodels[real_idx];
 
-    NPUW_ASSERT(m_compiled_submodels[idx].replaced_by);
+        // Due to concat some tensor should be skipped in closure
+        m_compiled_submodels[idx].closure.resize(0);
+        m_compiled_submodels[idx].is_remote.resize(0);
 
-    auto& comp_model_desc = m_compiled_submodels[idx];
+        for (std::size_t tidx = 0; tidx < comp_model_desc.transformations.size(); ++tidx) {
+            const auto& lt = m_compiled_submodels[idx].transformations[tidx];
```

Comment:
Why do you store transformations separately? shouldn't they accumulate somewhere within a LazyTensor?

Reply:
Done